### DISCUSSION TITLE
Support for subscription schedules

### DISCUSSION THREAD

**Discussion by hamdi777** [pay-rails/pay#961] [Ideas]
it would be so helpful if subscription schedules is integrated, many clients are demanding it , the idea is to be able to subscribe to multiple plans and these subscriptions get scheduled with the order of subscription , this is possible in stripe but not implemented in pay gem.

**Comment by KhanhVanCong** [comment]
I agree with you. In my opinion, we should consider subscription schedules as well. For instance, if users wish to downgrade from the 'Six-month plan' to the 'One-month plan,' they should be able to continue using the 'Six-month plan' until the end of the current billing cycle. After that, they can switch to the 'One-month plan' for their future subscriptions. This is a real case that I am stuck on it. `Pay` gem is cool gem but I hope that can support for these cases as well.

**Comment by cjilbert504** [comment]
At this moment, since Stripe is the only payment processor that supports this, we are not going to add this to Pay as there isn't really anything to abstract for all supported processors. For the time being, we suggest to use the code provided by the Stripe gem for this feature.
